metrics/firebase/functions/test/main_test.dart

```diff
@@ -0,0 +1,250 @@
+
+        final documentDataMatcher = predicate<DocumentData>((data) {
+          return data.getNestedData('totalDuration').getInt('operand') == 0;
+        });
+
+        final optionMatcher = predicate<SetOptions>((option) => option.merge);
+
+        verify(
+          _documentReferenceMock.setData(
+            argThat(documentDataMatcher),
+            argThat(optionMatcher),
+          ),
+        ).called(1);
+      },
+    );
+
+    test(
+      "uses a composite document id for the build days collection",
+      () async {
+        whenDocument(withCollectionName: buildDaysCollectionName)
+            .thenReturn(_documentReferenceMock);
+        whenSnapshotData().thenReturn(DocumentData.fromMap(build));
+
+        await onBuildAddedHandler(_documentSnapshotMock, _eventContextMock);
+
+        final documentId = '${projectId}_${utcDate.millisecondsSinceEpoch}';
+
+        verify(_collectionReferenceMock.document(documentId)).called(1);
+      },
+    );
+
+    test(
+      "creates a build days document from the build document snapshot's data",
+      () async {
+        whenDocumentData(withCollectionName: buildDaysCollectionName)
+            .thenAnswer((_) => Future.value());
+        whenSnapshotData().thenReturn(DocumentData.fromMap(build));
+
+        await onBuildAddedHandler(_documentSnapshotMock, _eventContextMock);
+
+        final documentDataMatcher = predicate<DocumentData>((data) {
+          final count =
+              data.getNestedData(buildDayStatusFieldName).getInt('operand');
+          final totalDuration =
+              data.getNestedData('totalDuration').getInt('operand');
+
+          final projectIdsAreEqual =
+              data.getString('projectId') == expectedBuildDayData['projectId'];
+          final countsAreEqual =
+              count == expectedBuildDayData[buildDayStatusFieldName];
+          final totalDurationAreEqual =
+              totalDuration == expectedBuildDayData['totalDuration'];
+          final daysAreEqual =
+              data.getTimestamp('day') == expectedBuildDayData['day'];
+
+          return projectIdsAreEqual &&
+              countsAreEqual &&
+              totalDurationAreEqual &&
+              daysAreEqual;
+        });
+
+        final optionMatcher = predicate<SetOptions>((option) => option.merge);
+
+        verify(
+          _documentReferenceMock.setData(
+            argThat(documentDataMatcher),
+            argThat(optionMatcher),
+          ),
+        ).called(1);
+      },
+    );
+
+    test("does not create a task document after created a build day", () async {
+      whenDocumentData(withCollectionName: buildDaysCollectionName)
+          .thenAnswer((_) => Future.value());
+      whenSnapshotData().thenReturn(DocumentData.fromMap(build));
+
+      await onBuildAddedHandler(_documentSnapshotMock, _eventContextMock);
+
+      verifyNever(_firestoreMock.collection(tasksCollectionName));
+    });
```

Comment:
It would be great to use a trailing coma here to be consistent with other tests.
```suggestion
    },);
```